src/Reader.php

```diff
@@ -52,6 +52,20 @@ public function query(callable $callable = null)
         $iterator = $this->applyIteratorFilter($iterator);
         $iterator = $this->applyIteratorSortBy($iterator);
         $iterator = $this->applyIteratorInterval($iterator);
+
```

Comment:
How about moving the code prior to `applyIteratorFilter` line ? This is an open question I don't know if it would change something or not ?